### PR TITLE
Added variable in .env to enforce certain domains in account registra…

### LINKED ISSUE #1548: enforce certain email-domains for registration
Just a feature-request - or maybe more something of an "I did something that I think might be useful for others, but I have no idea how to do a proper pull-request, so I just post the link to the change here".

Something that we needed (and I added in a "custom changes" fork/branch) was that only certain email-addresses were allowed for registration for greenlight accounts. We solved it by adding a variable in ".env" that can receive a list of domains that are checked on account-creation.

As mentioned above: I didn't find out how to create a pull-request that contains only one of my commits and not all of them, so please forgive me for just linking to the commit here. (One other commit would probably have been interesting, except that it recently got solved by somebody else: how to show only running rooms)

https://github.com/bigbluebutton/greenlight/commit/1e49876607a843184e552803aebb16fc53e6829c

Greetings and thanks for the great product,
~Emmy Graugans

### PR DESCRIPTION
**Pull request by EmmyGraugans**
suggestion for "enforce certain email-domains for registration" (Closes #1548)

## Description
Add an environment-variable `GREENLIGHT_ACCOUNT_HD` to enforce the "hosted domain" part of the Email-Address required for signup for a greenlight account

example:
GREENLIGHT_ACCOUNT_HD=@domain.com,subdomain-allowed.net

- subdomain-allowed.net matches also mail.subdomain-allowed.net
- @domain.com does *not* match @mail.domain.com

 multiple domains can be separated by comma (with no whitespace!)

## Testing Steps
1. Add the variable with one or more domains to ".env", restart Greenlight
1. Try to sign up for a greenlight-account with an unacceptable email -> get the error-message below
1. Try to sign up for a greenlight-account with an acceptable email -> the account will be created

## Screenshots (if appropriate):
error-message with unacceptable email-address
Bildschirmfoto 2021-04-08 um 09 23 18

### CODE REVIEW

**Comment by Sea444** [comment]
Would be great if this could be merged in upcoming 2.6. :-)
It's very useful if you have one greenlight frontend for several groups of users like schools. So you don't need allow/decline-option and can easily allow every user with certain mai ldomains.

**Comment by Sea444** [comment]
@EmmyGraugans I think you should delete german translation in your PR. Translations are made via Transifex. I can add new string in german translation as soon as PR is merged and english language file is automatically updates in Transifex.

**Comment by EmmyGraugans** [comment]
Sorry, took me a little while, first to find the time, then to find out, how to undo changes in a file. https://stackoverflow.com/questions/38743912/remove-a-file-from-a-git-pull-request#comment101821912_38744134 (the comment to the accepted answer) did the trick in the end.

**Comment by marcohald** [comment]
Is there any update to this pr ?

File: config/application.rb
Comment (farhatahmad) [inline config/application.rb:93]: This can probably be set to `config.require_email_domain = ENV["REQUIRE_MAIL_DOMAIN"].split(,)` (see below)

File: sample.env
Comment (farhatahmad) [inline sample.env:98]: 2 things: 
- Just for the sake of consistency, I'd prefer to name the ENV variable `LOCAL_ACCOUNT_HD` (or something along that sort)
- I would make it only support comma-separated strings `domain.com,subdomain-allowed.net`, since it would make everything else a lot cleaner

**Comment by marcohald** [comment]
@EmmyGraugans can you update your Pull Request ?
I would love to see this feature in upstream.

**Comment by EmmyGraugans** [comment]
I can try - my main-problem with updating my pull-request is that I don't quite understand what @farhatahmad means... For the sake of consistency with *what*? And what is LOCAL_ACCOUNT_HD supposed to mean? 

`grep -R LOCAL` gives exactly zero relevant hits in the greenlight tree
`grep -R ACCOUNT` shows that there exists a variable "ALLOW_GREENLIGHT_ACCOUNTS" which is at least kind of related.
`grep -R HD` turns up GOOGLE_OAUTH2_HD and OFFICE365_HD – ok, now I found at least something. 

The comment to GOOGLE_OAUTH2_HD reads "Google Apps hosted domains" but still doesn't give me an idea what a "hosted domain" is - isn't a Google-ID always in the domain google.com, gmail.com or googlemail.com?

Anyways, seeing now that HD indeed as something to do with domain, maybe consistent (with HD as in google and greenlight as in allow accounts) would be the variable-name `GREENLIGHT_ACCOUNT_HD` (and adjusting the comment above to include the words "hosted domain", so that people might get an idea what HD could mean) 

I will get to it tomorrow, or maybe late today. Before that, it's too busy here.

**Comment by EmmyGraugans** [comment]
Yesterday wasn't a good day, today was better ;)
I added the changes, but now it says something about `CI / Rubocop + RSpec (pull_request) Failing after 5m — Rubocop + RSpec` I'll try if I find a way to fix that (apparently the "Admin-changes-Email-address" field uses the same and causes problems, I need to find out how to apply the domain-check only on path "path=/b/signup" and not on "/b/admins/edit"...)

**Comment by farhatahmad** [comment]
The errors from Rspec were my mistake  - they're fixed now. Please update the branch from latest and the tests should pass

**Comment by marcohald** [comment]
@farhatahmad ist there anything missing on this pr or is it ready to get merged?

**Comment by farhatahmad** [comment]
Code looks good to me. Will come back to this in 2.8.1 (hopefully) and test it out

**Comment by farhatahmad** [comment]
Hi,

To help us better test and review community PRs, we have now implemented a **mandatory** PR template. We would really appreciate it if you were to update the description of this PR to match the template below. 

For more information, please see: https://github.com/bigbluebutton/greenlight#source-code--contributing

The template must be in the following format (it is automatically applied to all new PRs):
```
<!--- 
IMPORTANT
This template is mandatory for all Pull Requests.
Please follow the template to ensure your Pull Request is reviewed.
-->

<!--- Provide a general summary of your changes in the Title above -->

## Description
<!--- Describe your changes in detail -->

## Testing Steps
<!--- Please describe in detail how to test your changes. -->

## Screenshots (if appropriate):
<!--- Please include screenshots of ALL visual changes. -->

```

**Comment by EmmyGraugans** [comment]
I added the template to the description

**Review by farhatahmad — APPROVED** [review]

**Pull request merged**
